index.js

```diff
@@ -2,22 +2,39 @@
 var child = require('child_process');
 var execa = require('execa');
 var TEN_MEBIBYTE = 1024 * 1024 * 10;
+var porcelainRegex = /([AMRDC]|\?\?)\s/g;
 
-module.exports = function (dir) {
+function filterIgnored(stdout, ignored) {
+	var parsed = stdout.replace(porcelainRegex, '').trim().split('\n');
+
+	return parsed.filter(function (file) {
+		return ignored.indexOf(file) === -1;
+	});
+}
+
+module.exports = function (dir, ignored) {
```

Comment:
I would make it an options object. That way it's easy to add future options and it's much clearer what it is when you use it. But let's see what @jamestalmage think.